ir/type.cpp

```diff
@@ -635,10 +699,21 @@ expr SymbolicType::enforceAggregateType() const {
   return isArray() || isStruct();
 }
 
+expr SymbolicType::enforceFloatType() const {
+  return isFloat();
+}
+
 const StructType* SymbolicType::getAsStructType() const {
   return &s;
 }
 
+const FloatType* SymbolicType::getAsFloatType() const {
+  if (typ == Float)
```

Comment:
is this check needed?  why different than getAsStructType()?

Reply:
Here I want a class member function for class Type to check the given Type object is in FP type or not and returns a FloatType pointer. This function works similarly to the "dynamic_cast" and later a simple null check is able to tell whether a Type object is a FP.

In other word, I want the function to 
* return itself when it is a FloatType, 
* return member "f" when it is a SymbolicType and holds "typ == Float"
* returns nullptr otherwise. 

If we don't do this check, then SymbolicType objects will always returns a non-null pointer, even when the SymbolicType is not a FP type after fixup. This is not what it supposed to be.

Reply:
ok for now. The input functionality needs to be redesigned. I'll clean this up later.